test/e2e/windows/memory_limits.go

```diff
@@ -217,18 +202,34 @@ func getNodeMemory(ctx context.Context, f *framework.Framework) nodeMemory {
 	}
 
 	nodeMem := nodeMemory{
-		capacity:      status.Capacity[v1.ResourceMemory],
-		allocatable:   status.Allocatable[v1.ResourceMemory],
+		capacity:      node.Status.Capacity[v1.ResourceMemory],
+		allocatable:   node.Status.Allocatable[v1.ResourceMemory],
 		systemReserve: systemReserve,
 		hardEviction:  hardEviction,
-		// these are not implemented and are here for future use - will always be 0 at the moment
-		kubeReserve:  kubeReserve,
-		softEviction: softEviction,
+		kubeReserve:   kubeReserve,
+		softEviction:  softEviction,
 	}
 
 	return nodeMem
 }
 
+// getNodeMemory populates a nodeMemory struct with information from the first Windows node
+// that is found in the cluster.
+func getFirstNodeMemory(ctx context.Context, f *framework.Framework) nodeMemory {
+	selector := labels.Set{"kubernetes.io/os": "windows"}.AsSelector()
+	nodeList, err := f.ClientSet.CoreV1().Nodes().List(ctx, metav1.ListOptions{
+		LabelSelector: selector.String(),
+	})
+	framework.ExpectNoError(err)
+
+	// Assuming that agent nodes have the same config
+	// Make sure there is >0 agent nodes, then use the first one for info
+	gomega.Expect(nodeList.Items).ToNot(gomega.BeEmpty())
```

Comment:
We've had other tests fail becuase the nodes had different config (https://github.com/kubernetes/kubernetes/issues/106608).  This looks like it just moved logic but maybe need to consider opening an issue and fixing like we did for the other set of tests.